### ISSUE TITLE
Render planned path

### ISSUE THREAD

**Issue by ll-nick** [KIT-MRT/arbitration_graphs#37]
For the more complex behaviors it would be nice to see that path that's currently planned. Shouldn't be too hard but of course it's not exactly necessary either.

**Comment by ll-nick** [comment]
I just went about implementing the rendering like we talked about yesterday but there is another issue:

The "relative Path" does not contain any info about whether a tunnel is/was used. To infer whether the path goes through the tunnel, you also need info about the maze. Right now, this is not an issue because everything is neatly encapsulated into the environment model. But as soon as we move over to the PacmanWrapper, we cannot make use of the existing utility functions so easily.

I see a few options
1. Change the Command to just contain the absolute path and infer the SDL/Game command from that.
2. Have the PacmanAgent return both the Command struct and additionally the absolute path
3. Like 2 but wrap these types into another struct.
4. Just don't render the planned path and leave everything as it is right now.

What do you think @orzechow?

**Comment by orzechow** [comment]
:facepalm: 

>    1. Change the Command to just contain the absolute path and infer the SDL/Game command from that.

I would very much prefer a straightforward type directly designed for the agents environment. Thus, a single SDL Command aka direction would be best, but the current vector of directions is good enough as well (especially with its `Command(const Direction& direction)` constructor).

Also, converting to absolute coordinates would complicate the dumb single command behaviors which the tutorial will start with. Meaning, I really like the simpliness of e.g. `AvoidGhostBehavior::getCommand()` descriptive

```cpp
return Command{direction.value()};
```

or `MoveRandomlyBehavior`

```cpp
    int randomIndex = discreteRandomDistribution_(randomGenerator_);
    Move randomMove = Move::possibleMoves().at(randomIndex);

    return randomMove.direction;
```

Easy to understand, even with the caching stuff around it.

---

>     2. Have the PacmanAgent return both the Command struct and additionally the absolute path
> 
>     3. Like 2 but wrap these types into another struct.

These feel too clumsy for a tutorial.

---

>    4. Just don't render the planned path and leave everything as it is right now.

:sob: 

The path is really helpful to 
- highlight the difference between dumb single command vs path planning behaviors
- undestand the goal of the planning behaviors

---

After looking into the code, I guess passing an `EnvironmentModel::ConstPtr` to the pacman wrapper could be a minimal invasive compromise:

```cpp
demo.progressGame(command, agent.environmentModel());
```

**Comment by orzechow** [comment]
Btw. regarding our discussion on rendering a nice thick line with rounded corners instead of coloring the whole tiles:

- [Boost Geometry Buffer](https://www.boost.org/doc/libs/1_74_0/libs/geometry/doc/html/geometry/reference/algorithms/buffer/buffer_7_with_strategies.html) provides the geometric algorithm  
  ![](https://www.boost.org/doc/libs/1_74_0/libs/geometry/doc/html/img/algorithms/buffer_strategies.png)
- [`SDL_RenderGeometry`](https://wiki.libsdl.org/SDL2/SDL_RenderGeometry) provides the rendering function (see SDL3 docs for a [code example](https://wiki.libsdl.org/SDL3/SDL_RenderGeometry#code-examples))
- With [Boost Geometries Register Macros](https://www.boost.org/doc/libs/1_74_0/libs/geometry/doc/html/geometry/reference/adapted/register.html) we could directly use an `std::vector<SDL_Vertex>` when calling `boost::geometry::buffer()`

**Comment by orzechow** [comment]
Closed with #56
